tedge/src/cli/connect/command.rs

```diff
@@ -44,10 +45,21 @@ impl Cloud {
 
 impl Command for ConnectCommand {
     fn description(&self) -> String {
-        format!("create bridge to connect {} cloud.", self.cloud.as_str())
+        if self.is_test_connection {
+            format!("test connection to {} cloud.", self.cloud.as_str())
+        } else {
+            format!("create bridge to connect {} cloud.", self.cloud.as_str())
+        }
     }
 
     fn execute(&self, context: &ExecutionContext) -> Result<(), anyhow::Error> {
+        if self.is_test_connection {
+            match self.check_connection() {
+                Ok(()) => return Ok(()),
+                Err(e) => return Err(e.into()),
+            }
```

Comment:
In case of `Ok(())`, it needs to return `Ok(())` here to cancel the following codes (creating bridges,...). So, actually your suggestion and my code are not equal.